control-plane/subcommand/server-acl-init/command.go

```diff
@@ -306,8 +317,38 @@ func (c *Command) Run(args []string) int {
 		c.UI.Error(err.Error())
 	}
 
-	var bootstrapToken string
+	var backend SecretsBackend
+	switch SecretsBackendType(c.flagSecretsBackend) {
+	case SecretsBackendKubernetes:
+		backend = &KubernetesSecretsBackend{
+			ctx:          c.ctx,
+			clientset:    c.clientset,
+			k8sNamespace: c.flagK8sNamespace,
+			// TODO: should these use the global.acls.bootstrapToken.{secretName,secretKey}?
```

Comment:
I think I can't change this (yet) because may break existing deployments. 

1. If ACLs are pre-bootstrapped, then I would specify the k8s secret containing the bootstrap token in the Helm values, and server-acl-init [reads from the specified secret](https://github.com/hashicorp/consul-k8s/blob/main/charts/consul/templates/server-acl-init-job.yaml#L104-L110)
2. If server-acl-init automatically bootstrapped ACLs, then it writes the token to the hard-coded secret (`<prefix>-bootstrap-acl-token` name).

If we change the secret name, then server-acl-init would fail to find the bootstrap token in case 2. (I'm not clear how much of a problem this is if server-acl-init has already run once? But there are lots of upgrade scenarios to think through that it feels dangerous to do with my current limited understanding)

Reply:
To make k8s and Vault more consistent, I updated the secrets backends to use the given secret name, if provided, or else use a default secret name. This applies to both k8s and Vault.

I also removed the `-bootstrap-token-file` and the mounts/files that were used for the k8s and Vault secrets. Instead, the secret is read by server-acl-init directly from the secrets backend API (k8s or vault).

This should not be a breaking change because if a secret exists and contains the token, then that token is used (same as before). If the secret is empty, then we proceed with ACL bootstrapping and update the secret (previously, server-acl-init would error out).

Let me know what you think, and cc @thisisnotashwin for your thoughts on this